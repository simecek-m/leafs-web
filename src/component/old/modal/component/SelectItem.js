import React from "react";
import "component/old/modal/component/SelectItem.sass";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRight } from "@fortawesome/free-solid-svg-icons";

function SelectItem({ image = null, text = null, onClick = () => {} }) {
  const img = image ? (
    <img className="image" src={image} alt="select-item" />
  ) : null;
  return (
    <div className="select-item-component" onClick={onClick}>
      <span className="arrow">
        <FontAwesomeIcon icon={faArrowRight} />
      </span>
      {img}
      <span className="text">{text}</span>
    </div>
  );
}

export default SelectItem;
