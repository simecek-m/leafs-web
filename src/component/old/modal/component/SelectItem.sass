@import "colors.sass"

.select-item-component
  color: $background-primary
  text-align: left
  line-height: 35px
  height: 35px
  cursor: pointer
  font-size: 18px
  margin: 5px 0
  transition: 0.5s ease all

  .arrow
    display: inline-block
    vertical-align: top
    transition: 0.3s ease all
    height: 35px
    width: 0
    overflow: hidden
    text-align: center
    margin-right: 5px

  .image
    display: inline-block
    height: 35px
    width: 35px
    text-align: center
    transition: 0.3s ease all
    margin-right: 10px

  .text
    vertical-align: top
    transition: 0.3s ease all

  &:hover
    .arrow
      width: 35px

    .text
      font-weight: bold
