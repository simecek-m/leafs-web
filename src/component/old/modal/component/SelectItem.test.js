import React from "react";
import SelectItem from "component/old/modal/component/SelectItem";
import renderer from "react-test-renderer";
import csFlag from "assets/image/flag/cs.png";

const TEST_SELECT_ITEM_TEXT = "test";
const TEST_SELECT_ITEM_IMAGE = csFlag;

describe("modal SelectItem component", () => {
  test("should render default SelectItem", () => {
    const component = renderer.create(<SelectItem />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should render custom SelectItem", () => {
    const component = renderer
      .create(
        <SelectItem
          text={TEST_SELECT_ITEM_TEXT}
          image={TEST_SELECT_ITEM_IMAGE}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
