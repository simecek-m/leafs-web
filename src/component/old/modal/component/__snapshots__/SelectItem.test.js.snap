// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal SelectItem component should render custom SelectItem 1`] = `
<div
  className="select-item-component"
  onClick={[Function]}
>
  <span
    className="arrow"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-right fa-w-14 "
      data-icon="arrow-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <img
    alt="select-item"
    className="image"
    src="cs.png"
  />
  <span
    className="text"
  >
    test
  </span>
</div>
`;

exports[`modal SelectItem component should render default SelectItem 1`] = `
<div
  className="select-item-component"
  onClick={[Function]}
>
  <span
    className="arrow"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-right fa-w-14 "
      data-icon="arrow-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <span
    className="text"
  />
</div>
`;
